internal/controller/datadogagent/feature/controlplanemonitoring/feature.go

```diff
@@ -0,0 +1,206 @@
+	f.provider = provider
+	providerLabel, _ := kubernetes.GetProviderLabelKeyValue(provider)
+
+	// Add the writable emptyDir volume for all providers
+	agentConfDVolume := &corev1.Volume{
+		Name: emptyDirVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	managers.Volume().AddVolume(agentConfDVolume)
+
+	// Add volume mount to cluster-agent container
+	agentConfDVolumeMount := corev1.VolumeMount{
+		Name:      emptyDirVolumeName,
+		MountPath: controlPlaneMonitoringVolumeMountPath,
+		ReadOnly:  false,
+	}
+	managers.VolumeMount().AddVolumeMountToContainer(&agentConfDVolumeMount, apicommon.ClusterAgentContainerName)
+
+	// Select the appropriate configmap based on provider
+	var configMapName string
+	if providerLabel == kubernetes.OpenShiftProviderLabel {
+		configMapName = f.openshiftConfigMapName
+
+	} else if providerLabel == kubernetes.EKSProviderLabel {
+		configMapName = f.eksConfigMapName
+	} else {
+		configMapName = f.defaultConfigMapName
+	}
+	// Add the controlplane configuration configmap volume
+	if configMapName != f.defaultConfigMapName {
+		configMapVolume := &corev1.Volume{
+			Name: controlPlaneMonitoringVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: configMapName,
+					},
+				},
+			},
+		}
+		managers.Volume().AddVolume(configMapVolume)
+
+		// Add volume mount for the configmap
+		configMapVolumeMount := corev1.VolumeMount{
+			Name:      controlPlaneMonitoringVolumeName,
+			MountPath: controlPlaneMonitoringVolumeMountPath,
+			ReadOnly:  true,
+		}
+		managers.VolumeMount().AddVolumeMountToContainer(&configMapVolumeMount, apicommon.ClusterAgentContainerName)
+	}
+	return nil
+}
+
+// ManageSingleContainerNodeAgent allows a feature to configure the Agent container for the Node Agent's corev1.PodTemplateSpec
+// if SingleContainerStrategy is enabled and can be used with the configured feature set.
+// It should do nothing if the feature doesn't need to configure it.
+func (f *controlPlaneMonitoringFeature) ManageSingleContainerNodeAgent(managers feature.PodTemplateManagers, provider string) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *controlPlaneMonitoringFeature) ManageNodeAgent(managers feature.PodTemplateManagers, provider string) error {
+	f.provider = provider
+	return nil
+}
+
+// ManageClusterChecksRunner allows a feature to configure the ClusterChecksRunner's corev1.PodTemplateSpec
+func (f *controlPlaneMonitoringFeature) ManageClusterChecksRunner(managers feature.PodTemplateManagers) error {
+	providerLabel, _ := kubernetes.GetProviderLabelKeyValue(f.provider)
+	if providerLabel == kubernetes.OpenShiftProviderLabel {
+		etcdCertsVolume := &corev1.Volume{
```

Comment:
Why are we mounting this on CLC and not on DCA and creating config map mount in DCA and not here?